app/post/setup_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package post_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/grpc"
+	testkeyring "github.com/evmos/evmos/v15/testutil/integration/evmos/keyring"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/network"
+	evmtypes "github.com/evmos/evmos/v15/x/evm/types"
+	inflationtypes "github.com/evmos/evmos/v15/x/inflation/v1/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	gasLimit = 100_000
+)
+
+type PostTestSuite struct {
+	suite.Suite
+
+	unitNetwork *network.UnitTestNetwork
+	grpcHandler grpc.Handler
+	keyring     testkeyring.Keyring
+
+	txBuilder client.TxBuilder
+
+	from common.Address
+	to   common.Address
+}
+
+func (s *PostTestSuite) SetupTest() {
+	keyring := testkeyring.New(2)
+	unitNetwork := network.NewUnitTestNetwork(
+		network.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+	)
+	grpcHandler := grpc.NewIntegrationHandler(unitNetwork)
+
+	// TxBuilder is used to create Ethereum and Cosmos Tx to test
+	// the fee burner.
+	interfaceRegistry := codectypes.NewInterfaceRegistry()
+	codec := codec.NewProtoCodec(interfaceRegistry)
+	txConfig := authtx.NewTxConfig(codec, authtx.DefaultSignModes)
+	txBuilder := txConfig.NewTxBuilder()
+
+	s.from = keyring.GetAddr(0)
+	s.to = keyring.GetAddr(1)
+	s.unitNetwork = unitNetwork
+	s.grpcHandler = grpcHandler
+	s.keyring = keyring
+	s.txBuilder = txBuilder
+}
+
+func TestPostTestSuite(t *testing.T) {
+	suite.Run(t, new(PostTestSuite))
+}
+
+func (s *PostTestSuite) BuildEthTx() sdk.Tx {
+	chainID := s.unitNetwork.App.EvmKeeper.ChainID()
+	nonce := s.unitNetwork.App.EvmKeeper.GetNonce(
+		s.unitNetwork.GetContext(),
+		common.BytesToAddress(s.from.Bytes()),
+	)
+
+	ethTxParams := &evmtypes.EvmTxArgs{
+		ChainID:   chainID,
+		Nonce:     nonce,
+		To:        &s.to,
+		GasLimit:  gasLimit,
+		GasPrice:  big.NewInt(1),
+		GasTipCap: big.NewInt(1),
+	}
+
+	msgEthereumTx := evmtypes.NewTx(ethTxParams)
+	msgEthereumTx.From = s.from.String()
+	tx, err := msgEthereumTx.BuildTx(s.txBuilder, "evmos")
+	s.Require().NoError(err)
+	return tx
+}
```

Comment:
These kinds of functions could probably be added to the test utils, not sure what @facs95 thinks of it? Could be useful to be accessible in the unit testing framework but left out of the integration test setup maybe?

Reply:
maybe even use the `GenerateSignedEthTx` util? Not sure if the signature would interfere with anything here

https://github.com/evmos/evmos/blob/9cfbf0bc67c850303ab32d727609e2d707f80a0a/testutil/integration/evmos/factory/factory.go#L81-L93